gtsfm/bundle/bundle_adjustment.py

```diff
@@ -73,23 +73,25 @@ class BundleAdjustmentOptimizer:
 
     def __init__(
         self,
-        output_reproj_error_thresh: Optional[float] = None,
+        reproj_error_thresholds: List[Optional[float]] = [None],
         robust_measurement_noise: bool = False,
         shared_calib: bool = False,
         max_iterations: Optional[int] = None,
     ) -> None:
         """Initializes the parameters for bundle adjustment module.
 
         Args:
-            output_reproj_error_thresh (optional): the max reprojection error allowed in output. If the threshold is
-                                                   none, no filtering on output data is performed. Defaults to None.
+            reproj_error_thresholds (optional): List of reprojection error thresholds used to perform filtering after
```

Comment:
nit: I would remove the `optional` string in the docstring line here since to me, that suggest `Optional[List[float]]`, but instead the correct type is `List[Optional[float]]`